### ISSUE TITLE
epic: Compute authorization

### ISSUE THREAD

**Issue by davidfestal** [kcp-dev/kcp#2837]
### TL;DR

Define and implement how a user is allowed to see specific location workspaces and sync their workloads to them.

### Progress tracking <a href="#progress-tracking" id="progress-tracking">#</a>

EPIC detailed issues and overall progress can be tracked with the following [project view](https://github.com/orgs/kcp-dev/projects/1/views/32)

### Stories

- [ ] Design and implement compute authorization: what locations are avilable for a user
- [ ] Provide the user a way to see the locations which are available for him, with the corresponding labels, as well as the referenced APIExports
